manifests/install.pp

```diff
@@ -1,15 +1,15 @@
 # Install the foreman proxy
 class foreman_proxy::install {
   if ! $foreman_proxy::custom_repo {
-    foreman::repos { 'foreman_proxy':
+    class { '::foreman::repo':
```

Comment:
> I feel like we need some sort of puppet-foreman_common to manage this. Have puppet-foreman and puppet-foreman_proxy simply include it. If a user wants to change the repo, it should be done on the common module.

I suppose this is pretty much what we have with `foreman::repo`, just that it's stored within the puppet-foreman module.

If the repo/gpgcheck parameters were _removed_ from both puppet-foreman and puppet-foreman_proxy, we could argue that:

1. both puppet-foreman and puppet-foreman_proxy just `include ::foreman::repo`
1. the default behaviour for foreman::repo is to configure stable/gpgcheck=1
1. to customise either repo or gpgcheck, you set custom_repo=true on both foreman/foreman_proxy and include foreman::repo with the appropriate parameters - or use Hiera/data binding to set class parameters.

(I'm glossing over the custom_repo parameter on foreman::repo itself slightly.)